src/Sentry/ISpan.cs

```diff
@@ -66,6 +68,19 @@ public static ISpan StartChild(this ISpan span, string operation, string? descri
         return child;
     }
 
+    internal static ISpan StartChild(this ISpan span, SpanContext context)
+    {
+        var transaction = span.GetTransaction() as TransactionTracer;
+        if (transaction?.StartChild(context.SpanId, span.SpanId, context.Operation, context.Instrumenter)
+            is not SpanTracer childSpan)
+        {
+            return NoOpSpan.Instance;
```

Comment:
There are 4(?) different reasons why we can end up on this line.

The method is called `StartChild` on `ISpan` but there are a lot of assumptions/pre-requisites required for that to happen. And at the end it uses just `Description` from `SpanContext` which seem a bit obscure being buried in here.

Is this being called from `OnStart` and immediately casted to `SpanTracer`? If so, I think it belongs on that file, close together with the logic using/relying on its behavior.